src/main/java/com/google/devtools/build/lib/remote/RemoteModule.java

```diff
@@ -280,9 +280,12 @@ public void beforeCommand(CommandEnvironment env) throws AbruptExitException {
                 GoogleAuthUtils.newCallCredentials(authAndTlsOptions),
                 retrier);
         execChannel.release();
+        actionContextProvider = RemoteActionContextProvider.createForRemoteExecution(env,
+            (GrpcRemoteCache) cache, executor, executeRetrier, digestUtil, logDir);
```

Comment:
Maybe add a precondition that cache instanceof GrpcRemoteCache?

Reply:
done